Update release guide for maintainers (#323)
iterative/cml.dev: content/docs/contributing/core.md

  - Large PRs may be merged without squashing (but related commits should be
    squashed)
- **New releases**

>   - Run `gh workflow run release.yml --field bump=<increment>` where
>     `<increment>` is either `patch`, `minor` or `major`

Reviewer: ```suggestion
  - `gh workflow run release.yml --field bump={patch,minor,major}`
```

>   - Wait for an automatically created pull request to appear and approve it; it
>     will be merged as soon as checks succeed

Reviewer: ```suggestion
  - Wait for a PR to appear
  - Check & approve the PR
```

>   - Wait for the new release to appear, and customize the release notes if
>     required

Reviewer: needs to be `--draft`

```suggestion
  - Wait for a draft to appear under
    [releases](https://github.com/iterative/cml/releases)
  - Check & publish the draft
```

Author: That requires more human interaction 🤔

Reviewer: `--draft`: https://github.com/iterative/cml/pull/1165 :)
